lib/collector/remote-method.js

```diff
@@ -107,17 +107,22 @@ RemoteMethod.prototype._post = function _post(data, nrHeaders, callback) {
   }
 
   if (options.compressed) {
-    var useGzip = this._config.compressed_content_encoding === 'gzip'
-    var compressor = useGzip ? zlib.gzip : zlib.deflate
-    compressor(data, function onCompress(err, compressed) {
-      if (err) {
-        logger.warn(err, 'Error compressing JSON for delivery. Not sending.')
-        return callback(err)
-      }
-
-      options.body = compressed
-      makeRequest()
-    })
+    try {
+      var useGzip = this._config.compressed_content_encoding === 'gzip'
+      var compressor = useGzip ? zlib.gzip : zlib.deflate
+      compressor(data, function onCompress(err, compressed) {
+        if (err) {
+          logger.warn(err, 'Error compressing JSON for delivery. Not sending.')
+          return callback(err)
+        }
+
+        options.body = compressed
+        makeRequest()
+      })
+    } catch (err) {
+      logger.warn(err, 'Error compressing JSON for delivery. Not sending.')
+      return callback(err)
+    }
```

Comment:
Why the addition of the try/catch block? 

Did we have an errant throw? Where was it originating?

Reply:
I see through local experimentation that the `compressor` call is now throwing immediately with the invalid arg, instead of via the callback, when executed as a part of the remote method test "'should pass through error when compression fails."

This should work fine.

I might recommend leaving a comment as to why ('gzip and deflate now throw immediately in Node 14+ with an invalid argument' or similar).